core/tee/tee_svc_cryp.c

```diff
@@ -1464,8 +1489,19 @@ static TEE_Result tee_svc_cryp_obj_populate_type(
 		 * of the object
 		 */
 		if (type_props->type_attrs[idx].flags &
-		    TEE_TYPE_ATTR_SIZE_INDICATOR)
-			obj_size += attrs[n].content.ref.length * 8;
+			TEE_TYPE_ATTR_SIZE_INDICATOR) {
+		    /*
+		     * For ECDSA/ECDH we need to translate curve into
+		     * object size
+		     */
+			if (attrs[n].attributeID == TEE_ATTR_ECC_CURVE) {
+				res = get_ec_key_size(attrs[n].content.value.a,
+					&obj_size);
```

Comment:
previous implementation increased the `obj_size` while this change initializes it.
(minor: maybe add few tabulations to align function argument)